### ISSUE TITLE
DAG's parameter access_control is not refreshing in the UI

### ISSUE THREAD

**Issue by vdusek** [apache/airflow#8609]
**Apache Airflow version**: 1.10.9

**Environment**:

- **Cloud provider or hardware configuration**: N/A
- **OS** (e.g. from /etc/os-release): Debian GNU/Linux 9 (stretch)
- **Kernel** (e.g. `uname -a`): Linux 4cc8ac3c2cfb 5.4.0-26-generic #30-Ubuntu SMP Mon Apr 20 16:58:30 UTC 2020 x86_64 GNU/Linux
- **Install tools**: N/A

**What happened**:

When I update DAG's parameter `access_control` the change is not updated in the UI at `roles/list/`. I have to trigger refresh DAG manually in the UI to get the change. (I tried it many times and waited 10+ minutes.)

**What you expected to happen**:

I assume the change change should be updated automatically, as for any other DAG's parameter.

**How to reproduce it**:

Create new DAG for example with parameter: `access_control={'Public': ['can_dag_read']}`.

**Comment by iam432** [comment]
Even I see this issue, any update on this ticket, current workaround I am using to run airflow sync_perm command as cron job for every 5 minutes.

**Comment by vdusek** [comment]
@iam432 Can I ask you how do you use that command in the cron job? We use Airflow with Docker and I'm able to run `airflow sync_perm` inside the container, it works and the permissions are updated. However when I schedule that command (`* * * * * /usr/local/bin/airflow sync_perm`) in the cron, it doesn't work and I receive the following error:

```
[parameters: ('2020-05-06 14:20:02.886571', None, None, 'cli_sync_perm', None, 'airflow', '{"host_name": "857656ec757f", "full_command": "[\'/usr/local/bin/airflow\', \'sync_perm\']"}')]
(Background on this error at: http://sqlalche.me/e/e3q8)
The sync_perm command only works for rbac UI.
```

**Comment by iam432** [comment]
@vdusek : created a shell script for sync (there are other things in script) and scheduled cron run this script every 5 minutes as below.

[airflow@hostname ~]$ cat /data/airflow/sync.sh
#! /bin/bash
airflow sync_perm
[airflow@hostname ~]$

[airflow@hostname ~]$ crontab -l
*/5 * * * * sh /data/airflow/sync.sh
[airflow@hostname ~]$

**Comment by iam432** [comment]
@vdusek : forgot to say RBAC is enabled in my environment

**Comment by iam432** [comment]
@vdusek : Looks my solution is not working, did you find  a solution which is working for you ? please suggest if you have any workaround

**Comment by akotuc** [comment]
@iam432  Let me answer your question on behalf of @vdusek 

We noticed if you click on DAG refresh button on the right side of the main screen (in Links column) access rights are updated for the particular DAG - however you have to be admin.
This is not usable workaround since every time a new DAG is added someone with Admin rights would have to go into UI and press _refresh_ button.

At this moment we had to disable access control on the level of DAGs however it's something we need to resolve mid-term.

**Comment by jdavidheiser** [comment]
This is still an issue, and at least two more people have run into it (per discussion in Airflow Slack).  @feng-tao @astahlman per the comment on https://github.com/apache/airflow/blob/0eb5020fda46a6eceaa3652846598cb4ba34493b/airflow/www/security.py#L524, this seems like it should be a relatively simple fix to call `sync_perm_for_dag` for every dag when the rest of the permissions are refreshed.  Does that sound reasonable?  If so, I can submit a PR.

**Comment by feng-tao** [comment]
@jdavidheiser haven't got pinged in Airflow for a while :) If I recall(~ 2 years ago), we first need this commit (https://github.com/apache/airflow/pull/4642/files) to be use access_control in the Airflow version.

I have thought about that at that time to allow calling `sync_perm_for_dag` to work around this issue (Pinterest did this way). But the issue here is that RBAC lives in webserver/ flask-app builder while scheduler does the DAG parsing. Calling a webserver function within scheduler kinda violate the contract which ideally it should calls a DB function instead of a webserver internal call(scheduler shouldn't be aware of webserver internal). But the challenge is that all the RBAC db models live in RBAC instead of Airflow if I recall.

I think the ultimate solution is to have an API gateway(if we don't have that, webserver API is fine) which allows scheduler to calls that to update permission. What we did currently to workaround is to use cli to periodic update that part (cc @astahlman ). I haven't looked at the latest Airflow code for a while, will defer to @ashb or other to comment if there are other alternative approaches.

**Comment by gardnerdev** [comment]
Any update on this?

**Comment by brunoffaustino** [comment]
Any news on this issue?

**Comment by feng-tao** [comment]
cc @ashb @kaxil

**Comment by ashb** [comment]
Could someone try out alpha 2 and let us know if this is still a problem?

**Comment by kaxil** [comment]
Can some try it on 2.0.0beta3 @brunoffaustino @gardnerdev ?

**Comment by brunoffaustino** [comment]
Hi @kaxil and @ashb,
Me and @Cabeda just made the test under version 2.0.0beta3 and it still exists.
Here is how to we replicated it:
1. Install version 2.0.0beta3 via Docker;
2. Create another user called "test_user", besides having the "Admin" user;
3. Create role "test_role";
4. Remove all roles from user "test_user" and assign role "test_role" to it;
5. Change DAG "example_bash_operator" code to account with `access_control={'test_role': ['can_edit', 'can_read']},` inside the `DAG()` constructor;
6. See the DAG code change as an Admin logged in;
7. Log out and log in as "test_user";
8. **You don't see the DAG as it was expected**;
9. Log out and log in as "Admin";
10. Click the refresh button of the "example_bash_operator" DAG;
11. Log out and log in as "test_user";
12. Now you see the DAG.

Hope this helps tackling it. This one is really impacting our BAU when using Airflow.
If we can help in any other way. Let us know!

**Comment by potiuk** [comment]
Hey @ashb @kaxil @jhtimmins  -> I think that one is something we should - I think - fix for rc?

**Comment by alete89** [comment]
Also experiencing this bug on Airflow 1.10.12. The fact that you need to manually refresh de Dag from the UI kinda sucks.

**Comment by alexoliveira88** [comment]
I also have this bug in the version  1.10.14. The fact that you need to manually refresh de Dag from the UI.

**Comment by jedcunningham** [comment]
I'll be fixing this as part of #15311.

**Comment by czervenyvlk** [comment]
Thanks for your suggestions. running airflow sync_perm command with cron helps to mitigate this issue.
the bug has been reported already https://github.com/apache/airflow/issues/8609 and it should be fixed in 2.1 release.
We can close the ticket now

**Comment by alete89** [comment]
I'm Still experiencing this on 2.1.4 and 2.2.2
